src/components/form/adapters/base-adapter.ts

```diff
@@ -0,0 +1,211 @@
+import React from 'react';
+import { isNil, isNumber, isString, isBoolean } from 'lodash-es';
+
+const toDashCase = (myStr) =>
+    myStr.replace(/([A-Z])/g, (g) => `-${g[0].toLowerCase()}`);
+
+// Checks if value is a primitive
+const isPrimitive = (value) =>
+    isNil(value) || isNumber(value) || isString(value) || isBoolean(value);
+
+/**
+ * Renders a webcomponent by name
+ */
+export class LimeElementsAdapter extends React.Component<any, any> {
+    component = null;
+
+    static defaultProps = {
+        // The events to bind the the web component
+        events: {},
+        // The props to pass to the web component
+        elementProps: {},
+    };
+
+    constructor(
+        public props: {
+            name: string;
+            events?: any;
+            elementProps?: any;
+        }
+    ) {
+        super(props);
+
+        this.state = {
+            prevEvents: {},
```

Comment:
```suggestion
            previousEvents: {},
```

Reply:
As much as im for being more explicit with variables names, as i've reworked the code based on your suggestions, i use prev/next do identify previous props/events and the next props/events and its really pretty :rose: to look at cause they are both 4 characters :p So i would rather not do this but lemme know if you think its a really good idea :)

Reply:
Got here because I had to see what all the flowers were about 😄

I agree that the alignment of
```
nextSomething
prevSomething
```
warrants an exception to the rule of avoiding abbreviations 😊